### PR TITLE
Deprecate VectorView

### LINKED ISSUE #5278: Deprecate VectorView
`VectorView` is a class that creates an object that's derived from `Vector` by moving a external pointer under that class so that any memory location can be made to look like a `Vector<Number>`. In the process, it violates basically all rules of software design by accessing (parts) of the internal interface of the `Vector` class. I would like to see whether we can get rid of it.

Fortunately, there are only a couple of places in the library where it is being used. Here is one, in the matrix-free `operators.h` header:
```
        // copy the vector content to a temporary vector so that it does not get
        // lost
        VectorView<Number> view_src_in(subblock(src,i).local_size(),
                                       subblock(src,i).begin());
        const Vector<Number> copy_vec = view_src_in;
        subblock(const_cast<VectorType &>(src),i).
        reinit(data->get_dof_info(mf_component).vector_partitioner);
        VectorView<Number> view_src_out(subblock(src,i).local_size(),
                                        subblock(src,i).begin());
        static_cast<Vector<Number>&>(view_src_out) = copy_vec;
```
@kronbichler -- do you think this could easily be rewritten with something different now? It looks like we don't actually use much in terms of vector functionality, and that copying by hand (or via an `ArrayView` objects) would not be a major inconvenience (nor a performance issue). Do I see this right?

### PR DESCRIPTION
**Pull request by masterleinad**
Fixes #5278.

### CODE REVIEW

File: include/deal.II/lac/vector_view.h [hunk 1]
```diff
@@ -126,9 +126,11 @@ DEAL_II_NAMESPACE_OPEN
  * in the manual).
  *
  * @author Luca Heltai, 2009
+ *
+ * @deprecated Use ArrayView instead.
  */
 template <typename Number>
-class VectorView : public Vector<Number>
+class DEAL_II_DEPRECATED VectorView : public Vector<Number>
```
Comment (bangerth) [inline include/deal.II/lac/vector_view.h]: Oh, is this where it goes? I thought it needs to go to after the closing brace of the class and before the following semicolon. At least that's where we've had it in other cases.

Comment (davydden) [reply include/deal.II/lac/vector_view.h]: same here

Comment (masterleinad) [reply include/deal.II/lac/vector_view.h]: This is at least the correct position if we want to use `[[deprecated]]` in the future and it also works with `__attribute__((deprecated))`.
I'll switch to the other version regardless.for conformity.

**Comment by masterleinad** [comment]
Regarding the position of the deprecation attribute, https://godbolt.org/g/ASZVth shows that `[[deprecated]]` at the end of the class declaration is ignored.

**Comment by bangerth** [comment]
Let's merge this anyway since it's consistent with the placement elsewhere. Do you know which classes we currently deprecate, and should we move the marker consistently to between the `class` keyword and the class name?

**Pull request merged**